app/jobs/search_keyword_job.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class SearchKeywordJob < ApplicationJob
+  def perform(keyword_id)
+    keyword = Keyword.includes(:source).find(keyword_id)
+    keyword_name = keyword.name
+    search_result = search_service(keyword).new(keyword_name).call
```

Comment:
Can we just call the service here without the param?

We can see the duplication when passing the `keyword` instance and the `keyword_name` right after that. 

```suggestion
    search_result = search_service.new(keyword_name).call
```

So you can put the keyword to the instance variable

```
attr_reader :keyword
```

Then in the `def search_service` we can get it to detect the correspondence search service.

Reply:
```suggestion
    search_result = search_service).call
```

Good point Thanh! That makes me think that we can also initialize the service in the `search_service` method too ^^ 

(while Thanh suggestion makes total sense, consider my answer as optional - feel free to decline my suggestion ;-) )

Reply:
solved in ff0228d

Reply:
@phongvhd93 I think you put `.call`  at the wrong place: `search_result = search_service` feels wrong, it should be `search_result = search_service.call` (i.e. moving `.call` to the parent method, removing it from the `search_service` method as the later must return an instance of the search service 😇 )